src/services/codefixes/inferFromUsage.ts

```diff
@@ -0,0 +1,655 @@
+                            fileName: sourceFile.fileName,
+                            textChanges
+                        }]
+                    }];
+                }
+            }
+            return undefined;
+        }
+
+        function getCodeActionForSetAccessor(setAccessorDeclaration: SetAccessorDeclaration) {
+            const setAccessorParameter = setAccessorDeclaration.parameters[0];
+            if (!setAccessorParameter || !isIdentifier(setAccessorDeclaration.name) || !isIdentifier(setAccessorParameter.name)) {
+                return undefined;
+            }
+
+            const type = inferTypeForVariableFromUsage(setAccessorDeclaration.name) ||
+                inferTypeForVariableFromUsage(setAccessorParameter.name);
+            const typeString = type && typeToString(type, containingFunction);
+            if (!typeString) {
+                return undefined;
+            }
+
+            return createCodeActions(setAccessorDeclaration.name.getText(), setAccessorParameter.name.getEnd(), `: ${typeString}`);
+        }
+
+        function getCodeActionForGetAccessor(getAccessorDeclaration: GetAccessorDeclaration) {
+            if (!isIdentifier(getAccessorDeclaration.name)) {
+                return undefined;
+            }
+
+            const type = inferTypeForVariableFromUsage(getAccessorDeclaration.name);
+            const typeString = type && typeToString(type, containingFunction);
+            if (!typeString) {
+                return undefined;
+            }
+
+            const closeParenToken = getFirstChildOfKind(getAccessorDeclaration, sourceFile, SyntaxKind.CloseParenToken);
+            return createCodeActions(getAccessorDeclaration.name.getText(), closeParenToken.getEnd(), `: ${typeString}`);
+        }
+
+        function createCodeActions(name: string, start: number, typeString: string) {
+            return [{
+                description: formatStringFromArgs(getLocaleSpecificMessage(Diagnostics.Infer_type_of_0_from_usage), [name]),
+                changes: [{
+                    fileName: sourceFile.fileName,
+                    textChanges: [{
+                        span: { start, length: 0 },
+                        newText: typeString
+                    }]
+                }]
+            }];
+        }
+
+        function getReferences(token: PropertyName | Token<SyntaxKind.ConstructorKeyword>) {
+            const references = FindAllReferences.findReferencedSymbols(
+                program,
+                cancellationToken,
+                program.getSourceFiles(),
+                token.getSourceFile(),
+                token.getStart());
+
+            Debug.assert(!!references, "Found no references!");
+            Debug.assert(references.length === 1, "Found more references than expected");
+
+            return map(references[0].references, r => <Identifier>getTokenAtPosition(program.getSourceFile(r.fileName), r.textSpan.start, /*includeJsDocComment*/ false));

```

Comment:
references might be in jsdoc comments, so getTokenAtPosition might not be an identifier. and might in fact be undefined? (probably not in typescript, but perhaps in Javascript)

Reply:
can not get this to barf in a .ts file. i will have to revisit that when i add the .js part to it. Added a check to the top of the function to ensure we are do not offer the quickfix in a .js file